osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -29,32 +28,44 @@ public GameplayLeaderboard()
         /// <summary>
         /// Adds a player to the leaderboard.
         /// </summary>
-        /// <param name="currentScore">The bindable current score of the player.</param>
+        /// <param name="totalScore">A bindable of the player's total score.</param>
+        /// <param name="accuracy">A bindable of the player's accuracy.</param>
+        /// <param name="combo">A bindable of the player's current combo.</param>
         /// <param name="user">The player.</param>
-        public void AddPlayer([NotNull] BindableDouble currentScore, [NotNull] User user)
+        public void AddPlayer([NotNull] IBindableNumber<double> totalScore,
+                              [NotNull] IBindableNumber<double> accuracy,
+                              [NotNull] IBindableNumber<int> combo,
+                              [NotNull] User user)
```

Comment:
i'd just make `GameplayLeaderboardScore` a non-nested public model class at this point and just pass that in instead of jamming all 4 of its constituent members through 2 levels of calls.

could also make a ctor that just takes the user in, and initialises all other bindables to sane initial values. would clean up the test helpers.

Reply:
Sure that'll work.